smslibrary/src/test/java/com/eis/smslibrary/SMSManagerTest.java

```diff
@@ -36,14 +36,13 @@
     @Captor
     private ArgumentCaptor<String> peerNumberCaptor;
 
-    private SmsManager managerMock = mock(SmsManager.class);
+    private final SmsManager MANAGER_MOCK = mock(SmsManager.class);
 
     private final String VALID_PEER_NUMBER = "+393423541601";
     private final String VALID_MESSAGE_TEXT = "valid text";
-    private final String SENT_TEXT = (char) 0x02 + VALID_MESSAGE_TEXT;
     private final SMSPeer VALID_PEER = new SMSPeer(VALID_PEER_NUMBER);
     private final SMSMessage VALID_MESSAGE = new SMSMessage(VALID_PEER, VALID_MESSAGE_TEXT);
-    private SMSManager managerInstance = SMSManager.getInstance();
+    private final SMSManager MANAGER_INSTANCE = SMSManager.getInstance();
```

Comment:
`final` marks constants
also if a variable is marked as `final`, every object instance of the class sees that variable in the same way, effectively making it static

Reply:
>final marks constants

This is not true, from the official Java Coding Style Guide I linked:
> The
following are considered to be constants:
>1. All static final primitive types (Remember that all interface fields are inherently static
final).
>2. All static final object reference types that are never followed by "." (dot).
>3. All static final arrays that are never followed by "[" (dot).

As you can see, final fields that are not also static are not considered constants and should not be all upper-case.

Reply:
I don't feel like arguing over it but Giovanni has a point. I personally only put in full capital both `static` and `final` fields.

Reply:
The only difference between a final variable and a static final variable is this:
```
public class Foo{
    public final int a = 5;
    public static final int b = 10;
}

//outside method
public void barMethod(){
    Foo fooInstance = new Foo();
    fooInstance.a; // returns 5
    Foo.b; //returns 10
    fooInstance.a = 20; //ERROR
    Foo.b = 30; //ERROR
}
```

So yeah, the only difference is that you can access the `static final` variable without needing to instantiate the class, while you need that instance to get that `final` variable.

But in our case that variable is private, so it can't be accessed from outside, making this difference irrelevant, so effectively making it a constant

Reply:
> But in our case that variable is private, so it can't be accessed from outside, making this difference irrelevant, so effectively making it a constant

No, the definition of "constant" does not contemplate object references without the `static` keyword. Even if `MANAGER_INSTANCE` is initialized when the class is loaded, just like a constant, it cannot be a constant because of its definition.

Reply:
Yeah ok, I get it, you only want to follow the definition, without thinking abuot the problem itself, alright, I'll make the change, there's no point in arguing for this.
But to me that's still a constant:
[If it looks like a duck, swims like a duck, and quacks like a duck, then it probably is a duck.](https://en.wikipedia.org/wiki/Duck_test)

Reply:
Even when you look at how a `private final` field and a `private static final` variable are treated by the JVM they're different, a `static` variable is initialized only once at the start of execution by the ClassLoader, while a `private final` field is initialized every time an instance of the object is created.